src/nomad_simulations/outputs.py

```diff
@@ -140,23 +105,52 @@ class SCFOutputs(Outputs):
         """,
     )
 
-    is_scf_converged = Quantity(
-        type=bool,
-        description="""
-        Flag indicating whether the output property is converged or not after a SCF process. This quantity is connected
-        with `SelfConsistency` defined in the `numerical_settings.py` module.
-        """,
-    )
-
-    self_consistency_ref = Quantity(
-        type=SelfConsistency,
-        description="""
-        Reference to the `SelfConsistency` section that defines the numerical settings to converge the
-        output property.
-        """,
-    )
+    def resolve_is_derived(self, logger: BoundLogger) -> bool:
+        """
+        Resolves if the physical property is converged or not after a SCF process.
 
-    # TODO add more functionality to automatically check convergence from `self_consistency_ref` and the last two `scf_steps`
+        Returns:
+            (bool): The flag indicating whether the physical property is converged or not after a SCF process.
+        """
+        for property_name in self.m_def.all_sub_sections.keys():  # ! Check this
+            # We get the property of `SCFOutputs` that we want to check
+            prop = getattr(self, property_name)
+
+            # If it is None or does not contain a `self_consistency_ref` quantity we continue
+            if prop is None:
+                continue
+            scf_parameters = prop.self_consistency_ref
+            if scf_parameters is None:
+                continue
```

Comment:
Why are these checks needed, if only `self.scf_steps` is used to verify convergence?